torchvision/prototype/datasets/video_utils.py

```diff
@@ -0,0 +1,192 @@
+from typing import Any, Callable, Dict, List, Optional, Tuple, Mapping, Union, Iterator
+import random
+import av
+import numpy as np
+import torch
+from torchdata.datapipes.iter import IterDataPipe
+from torchvision.io import video, _video_opt
+
+class AVKeyframeReader(IterDataPipe[Dict[str, Any]]):
+    def __init__(self, video_dp: IterDataPipe[Dict[str, Any]], transform=None) -> None:
+        """TorchData Iterdatapype that takes in video datapipe
+        and yields all keyframes in a video
+
+        Args:
+            video_dp (IterDataPipe[Dict[str, Any]]): Video dataset IterDataPipe
+            transform ([type], optional): Transform to be applied to every frame. Defaults to None.
+        """
+        self.datapipe = video_dp
+        self.transform = transform
```

Comment:
That is an anti-pattern we want to avoid. Transformations can applied in a separate `.map` call.

Reply:
will remove it - force of habit I suppose :)